bin/registerServerTopology-Func.php

```diff
@@ -55,14 +55,124 @@ function formatResponseMessage(int $code, string $message, string $type = 'succe
     switch ($type) {
         case 'success':
             $responseMessage = 'code: ' . $code . PHP_EOL .
-            'message: ' . $message . PHP_EOL;
+                'message: ' . $message . PHP_EOL;
             break;
         case 'error':
         default:
             $responseMessage = 'error code: ' . $code . PHP_EOL .
-            'error message: ' . $message . PHP_EOL;
+                'error message: ' . $message . PHP_EOL;
             break;
     }
 
     return sprintf('%s', $responseMessage);
 }
+
+/**
+ * Parse the template to an array usable by the script
+ *
+ * @param string $path
+ * @return array
+ */
+function parseTemplateFile(string $path): array
+{
+    $data = file_get_contents($path);
+    //Remove the blank line
+    $data = preg_replace("/(^[\r\n]*|[\r\n]+)[\s\t]*[\r\n]+/", "\n", $data);
+    $lines = preg_split("/(\n)/", $data);
+    $configVariables = [];
+    foreach ($lines as $line) {
+        if (preg_match('/^(.+?)=(.+)$/', $line, $match)) {
+            $configVariables[trim($match[1])] = trim($match[2]);
+        }
+    }
+    return castTemplateValue($configVariables);
+}
+
+/**
+ * Cast correct type for each values
+ *
+ * @param array $configVariables
+ * @return array
+ */
+function castTemplateValue(array $configVariables): array
+{
+    foreach ($configVariables as $configKey => $configValue) {
+        switch ($configKey) {
+            case 'INSECURE':
+            case 'PROXY_USAGE':
+                $configVariables[$configKey] = (bool) $configValue;
+                break;
+            case 'PROXY_PORT':
+                $configVariables[$configKey] = (int) $configValue;
+                break;
+        }
+    }
+    return $configVariables;
+}
+
+/**
+ * Refactor all the settings options mechanism to avoid duplication in the script part
+ *
+ * @param array $options
+ * @param string $helpMessage
+ * @return array
+ */
+function setConfigOptionsFromTemplate(array $options, string $helpMessage): array
+{
+    $configOptions = [];
+    if (
+        !isset(
+            $options['API_USERNAME'],
+            $options['API_PASSWORD'],
+            $options['SERVER_TYPE'],
+            $options['HOST_ADDRESS'],
+            $options['SERVER_NAME']
+        )
+    ) {
+        throw new \InvalidArgumentException(
+            PHP_EOL .
+                'missing value: API_USERNAME, API_PASSWORD, SERVER_TYPE, HOST_ADDRESS and SERVER_NAME are mandatories'
+                . PHP_EOL . $helpMessage
+        );
+    }
+
+    $configOptions['API_USERNAME'] = $options['API_USERNAME'];
+    $configOptions['SERVER_TYPE'] = in_array(strtolower($options['SERVER_TYPE']), SERVER_TYPES)
```

Comment:
Mixing the types of variables is not recommended.
Instead of using the value 'false' use 'null' instead.